sktime/utils/_testing/scenarios.py

```diff
@@ -262,11 +262,10 @@ def _check_dict_of_dict(obj, name="obj"):
     ------
     TypeError if obj is not dict of dict, with str keys
     """
-    msg = f"{obj} must be a dict of dict, with str keys"
-    if not isinstance(obj, dict):
-        raise TypeError(msg)
-    if not np.all([isinstance(x, dict) for x in obj.values()]):
-        raise TypeError(msg)
-    if not np.all([isinstance(x, str) for x in obj.keys()]):
-        raise TypeError(msg)
+    if not (
+        isinstance(obj, dict)
+        and np.all([isinstance(x, dict) for x in obj.values()])
+        and np.all([isinstance(x, str) for x in obj.keys()])
+    ):
```

Comment:
```suggestion
    if not (
        isinstance(obj, dict)
        and all(isinstance(x, dict) for x in obj.values())
        and all(isinstance(x, str) for x in obj.keys())
    ):
```

Suggesting 2 changes:

1. `np.all` to `all`
2. allow short-circuit

I'm suggesting from Github site, code's not tested.